pkg/odo/cli/url/describe.go

```diff
@@ -38,16 +42,21 @@ type URLDescribeOptions struct {
 	componentContext string
 	url              string
 	*genericclioptions.Context
+	devfilePath string
+	isDevFile   bool
 }
 
 // NewURLDescribeOptions creates a new NewURLDescribeOptions instance
 func NewURLDescribeOptions() *URLDescribeOptions {
-	return &URLDescribeOptions{&config.LocalConfigInfo{}, "", "", &genericclioptions.Context{}}
+	return &URLDescribeOptions{&config.LocalConfigInfo{}, "", "", &genericclioptions.Context{}, "", false}
 }
 
 // Complete completes URLDescribeOptions after they've been Listed
 func (o *URLDescribeOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	if experimental.IsExperimentalModeEnabled() {
+	o.devfilePath = clicomponent.DevfilePath
```

Comment:
Join the filepath with o.componentContext otherwise context won't work
```
[mrinaldas@localhost project]$ odo url describe example-11
I0903 12:31:27.064141   17431 cert_rotation.go:137] Starting client certificate rotation controller
NAME           STATE          URL                       PORT     SECURE     KIND
example-11     Not Pushed     http://example-11.com     3000     false      ingress
There are local changes. Please run 'odo push'.
[mrinaldas@localhost project]$ cd ..
[mrinaldas@localhost ~]$ odo url describe example-11 --context project
I0903 12:31:43.739120   17482 cert_rotation.go:137] Starting client certificate rotation controller
 ✗  The current directory does not represent an odo component. Use 'odo create' to create component here or switch to directo...
```

Reply:
updated